### PR TITLE
Groundwork for Best Practices content.

### PR DESCRIPTION
**Pull request by Serdaro**
* Collecting all best practices (not the ones talking about improving query performance for now) under one heading within "Deploy Clusters" chapter
* Mindmap: https://miro.com/app/board/uXjVKVdhIQ8=/?share_link_id=944410513893
* Additional helpful resource benefited from: https://docs.google.com/presentation/d/1c1PFTg3MehvUfaBLyjIRy5qd8op7P7-t/edit?pli=1#slide=id.p1
* HD and TPC leveled up, having their own sections

### CODE REVIEW

File: docs/modules/cluster-performance/pages/threading.adoc [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+-Dhazelcast.client.io.output.thread.affinity=5-8
+-Dhazelcast.client.response.thread.affinity=7-9
+```
+
+You can set the CPU thread affinity properties shown above only on the command line. 
+
+Let's have a look at how we define the values for the above configuration
+properties:
+
+* **Individual CPUs**, e.g., `1,2,3`: This means there are going to be
```
Comment (rebekah-lawrence) [inline docs/modules/cluster-performance/pages/threading.adoc:216]:
```suggestion
* **Individual CPUs**, for example `1,2,3`: This means there are going to be
```

File: docs/modules/cluster-performance/pages/threading.adoc [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+three threads. The first thread runs on CPU 1, the second thread on CPU 2, and so on.
+* **CPU ranges**, e.g., `1-3`: Shortcut syntax for `1,2,3`.
```
Comment (rebekah-lawrence) [inline docs/modules/cluster-performance/pages/threading.adoc:218]:
```suggestion
* **CPU ranges**, for example `1-3`: Shortcut syntax for `1,2,3`.
```

File: docs/modules/cluster-performance/pages/threading.adoc [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+* **Group**, e.g., `[1-3]`: This configures three threads and each of
```
Comment (rebekah-lawrence) [inline docs/modules/cluster-performance/pages/threading.adoc:219]:
```suggestion
* **Group**, for example `[1-3]`: This configures three threads and each of
```

File: docs/modules/cluster-performance/pages/threading.adoc [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+these threads can run on CPU 1, 2 and 3.
+* **Group with thread count**, e.g., `[1-3]:2`: This configures two
```
Comment (rebekah-lawrence) [inline docs/modules/cluster-performance/pages/threading.adoc:221]:
```suggestion
* **Group with thread count**, for example `[1-3]:2`: This configures two
```

File: docs/modules/cluster-performance/pages/threading.adoc [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+threads and each of these two threads can run on CPU 1, 2 and 3.
+
+You can also combine those, e.g., `1,2,[5-7],[10,12,16]:2`.
```
Comment (rebekah-lawrence) [inline docs/modules/cluster-performance/pages/threading.adoc:224]:
```suggestion
You can also combine those, for example `1,2,[5-7],[10,12,16]:2`.
```

File: docs/modules/cluster-performance/pages/threading.adoc [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+
+Note that, the syntax for CPU thread affinity shown above not only determines
+the mapping of CPUs to threads, it also determines the thread count.
+If you use CPU thread affinity, e.g., `hazelcast.io.input.thread.affinity`,
```
Comment (rebekah-lawrence) [inline docs/modules/cluster-performance/pages/threading.adoc:228]:
```suggestion
If you use CPU thread affinity - for example, `hazelcast.io.input.thread.affinity` - 
```

File: docs/modules/cluster-performance/pages/threading.adoc [hunk 7]
```diff
@@ -0,0 +1,338 @@
...
+```
+
+If you want to configure 20 threads on NUMA node 0 and 20 threads on NUMA node 1,
+and confine the threads to these NUMA nodes, you can use the following configuration:
+
+```
+-Dhazelcast.operation.thread.affinity=[0-9,20-29],[10-19,30-39]
+```
+
+See https://en.wikipedia.org/wiki/Non-uniform_memory_access[here^]
```
Comment (rebekah-lawrence) [inline docs/modules/cluster-performance/pages/threading.adoc:265]:
```suggestion
See https://en.wikipedia.org/wiki/Non-uniform_memory_access[here^]
```
```suggestion
Refer to https://en.wikipedia.org/wiki/Non-uniform_memory_access[Non-uniform memory access^]
```

File: docs/modules/cluster-performance/pages/threading.adoc [hunk 8]
```diff
@@ -0,0 +1,338 @@
...
+
+The `SlowOperationDetector` monitors the operation threads and collects information about all slow operations.
+An `Operation` is a task executed by a generic or partition thread (see xref:performance:threading-model.adoc#operation-threading[Operation Threading]).
+An operation is considered as slow when it takes more computation time than the configured threshold.
+
+The `SlowOperationDetector` stores the fully qualified classname of the operation and its stacktrace as well as
+operation details, start time and duration of each slow invocation. All collected data is available in
+the xref:{page-latest-supported-mc}@management-center:monitor-imdg:monitor-members.adoc[Management Center].
+
+The `SlowOperationDetector` is configured via the following system properties.
```
Comment (rebekah-lawrence) [inline docs/modules/cluster-performance/pages/threading.adoc:278]:
```suggestion
The `SlowOperationDetector` is configured using the following system properties.
```

File: docs/modules/cluster-performance/pages/threading.adoc [hunk 9]
```diff
@@ -0,0 +1,338 @@
...
+You can set the response thread for internal operations both on the members and clients.
+By setting the backoff mode on and depending on the use case, you can get a
+5-10% performance improvement. However, this increases the CPU utilization.
+To enable backoff mode please set the following property for Hazelcast cluster members: 
+
+```
+-Dhazelcast.operation.responsequeue.idlestrategy=backoff  
+```
+
+For Hazelcast clients, please use the following property to enable backoff:
```
Comment (rebekah-lawrence) [inline docs/modules/cluster-performance/pages/threading.adoc:329]:
```suggestion
For Hazelcast clients, use the following property to enable backoff:
```

**Review by rebekah-lawrence — COMMENTED** [review]
Kept coming back over the course of the day.....some suggestions for your consideration.

File: docs/modules/cluster-performance/pages/performance-tips.adoc [hunk 10]
```diff
@@ -475,6 +655,16 @@ be sure to profile the data load time and to take that startup time into account
 
 See the xref:storage:high-density-memory.adoc[HDMS section] to learn more.
 
+== Cluster Size
+
+Here are the essential tips:
+
+* Split-brain is a network break, it affects hosts
+* You can’t stop a network from a psychical or logical break
```
[the remark this answers is not in the document]
Comment (Serdaro) [reply docs/modules/cluster-performance/pages/performance-tips.adoc:665]: yeah.. the original word looks strange :)))

[the remark this answers is not in the document]
Comment (Serdaro) [reply docs/modules/cluster-performance/pages/performance-tips.adoc]: https://github.com/hazelcast/hz-docs/pull/1079/commits/c815cfa3fd69f3c8976e2086cfbf2d843060785e

File: docs/modules/cluster-performance/pages/near-cache.adoc [hunk 11]
```diff
@@ -0,0 +1,656 @@
...
+* `in-memory-format`: Specifies in which format data is stored in your Near Cache.
+Note that a map's in-memory format can be different from that of its Near Cache. Available values are as follows:
+** `BINARY`: Data is stored in serialized binary format (default value).
+** `OBJECT`: Data is stored in deserialized form.
+** `NATIVE`: Data is stored in the Near Cache that uses Hazelcast's High-Density Memory Store feature.
+This option is available only in Hazelcast Enterprise. Note that a map and
+its Near Cache can independently use High-Density Memory Store.
+For example, while your map does not use High-Density Memory Store, its Near Cache can use it.
+* `invalidate-on-change`: Specifies whether the cached entries are evicted when the entries are updated or removed.
+Its default value is true.
```
[the remark this answers is not in the document]
Comment (Serdaro) [reply docs/modules/cluster-performance/pages/near-cache.adoc]: Yes. https://github.com/hazelcast/hz-docs/pull/1079/commits/c815cfa3fd69f3c8976e2086cfbf2d843060785e

**Comment by Serdaro** [comment]
@yozank , @wyohm , in addition to the best practices improvements journey here, especially the above commit (https://github.com/hazelcast/hz-docs/pull/1079/commits/7b30b3c9e93a126b876814bee3b77bffef743d9f) is for your review.

**Review by wyohm — APPROVED** [review]
Map entry section looks good.

**Pull request closed without merging**